Modules/Filtering/ImageFrequency/include/itkFrequencyBandImageFilter.hxx

```diff
@@ -134,102 +137,83 @@ FrequencyBandImageFilter< TImageType, TFrequencyIterator >
 template< typename TImageType, typename TFrequencyIterator >
 void
 FrequencyBandImageFilter< TImageType, TFrequencyIterator >
-::DynamicThreadedGenerateData(
-  const ImageRegionType & outputRegionForThread)
+::BandPass( FrequencyIteratorType& freqIt )
 {
-  const ImageType *inputPtr = this->GetInput();
-  ImageType *outputPtr = this->GetOutput();
-
-  // Define the portion of the input to walk for this thread
-  ImageRegionType inputRegionForThread;
-  this->CallCopyOutputRegionToInputRegion(inputRegionForThread, outputRegionForThread);
-  // copy the input pixel to the output
-  ImageAlgorithm::Copy( inputPtr, outputPtr, inputRegionForThread, outputRegionForThread );
-
-  FrequencyIteratorType freqIt(outputPtr, outputRegionForThread);
-  // Only for HalfHermitian Iterator this option changes frequency spacing
-  freqIt.SetActualXDimensionIsOdd(this->GetActualXDimensionIsOdd());
-
-  freqIt.GoToBegin();
   FrequencyValueType f;
   typename FrequencyIteratorType::FrequencyType w;
   FrequencyValueType wMax;
   FrequencyValueType wMin;
   bool freqIsNegative = false;
 
-  while ( !freqIt.IsAtEnd() )
+  if ( this->m_RadialBand )
     {
-    if ( this->m_RadialBand )
+    f = sqrt(freqIt.GetFrequencyModuloSquare());
```

Comment:
`f" -> `frequency`

Please use descriptive names, non-single character names, that follow the ITK style guidelines for readability and so the codebase is searchable.